src/main/java/com/percero/agents/sync/access/RedisAccessManager.java

```diff
@@ -566,29 +565,40 @@ public void destroyClient(String clientId) {
 	 * This function checks to see if a User has no more UserDevices. If they do not have
 	 * 	any, then it removes all the User's entries in ALL AccessJournal sets.
 	 * 
-	 * @param userId
-	 * @param clientId
+    	 * @param clientId
 	 */
 	@SuppressWarnings("unchecked")
 	@Transactional
 	private void deleteClientAccessJournals(String clientId) {
 		String clientAccessJournalsKey = RedisKeyUtils.clientAccessJournal(clientId);
-		Set<String> clientAccessJournals = (Set<String>) redisDataStore.getSetValue(clientAccessJournalsKey);
-		redisDataStore.removeSetsValue(RedisKeyUtils.ACCESS_JOURNAL_PREFIX, clientAccessJournals, clientId);
+		Set<String> clientAccessJournals = (Set<String>) cacheDataStore.getSetValue(clientAccessJournalsKey);
+		cacheDataStore.removeSetsValue(RedisKeyUtils.ACCESS_JOURNAL_PREFIX, clientAccessJournals, clientId);
 		
 		// Now remove the Client's AccessJournal key.
-		redisDataStore.deleteKey(clientAccessJournalsKey);
+		cacheDataStore.deleteKey(clientAccessJournalsKey);
+
+        // Check to see if each Object's access journal set is empty and if so remove it
+        // from the class access journal set
+        for(String caj : clientAccessJournals){
+            if(cacheDataStore.getSetIsEmpty(RedisKeyUtils.ACCESS_JOURNAL_PREFIX+caj)){
+                String[] parts = caj.split(":");
```

Comment:
We do this kind of thing a lot, you should probably refactor that into a helper method.  Add it to the ClassIDPair class like
```java
public static ClassIDPair fromCacheKey(String cacheKey){
  String[] parts = cacheKey.split(":");
  className = parts[0];
  ID = parts[1];
}
```